### ISSUE TITLE
CMake's parallel option not respected when using a build preset

### ISSUE THREAD

**Issue by vadi2** [microsoft/vscode-cmake-tools#3176]
### Brief Issue Summary

I'm getting into the world of build presets, and found that my VSCode configuration - such as the number of parallel builds - is not respected when using a specific build preset.

I'd like it very much to still be respected and take precedence over any presets. I'm aware user presets are an option in cmake, but unless you're going to be stashing and popping them all the time, it is just not practical to store user configuration in the repo code.

As a workaround, using the 'default' build preset is respecting the option so that is nice!

**Comment by benmcmorran** [comment]
Does it work for you to `.gitignore` your `CMakeUserPresets.json`? My understanding is that that's how the file is designed to be used, so that you can have custom user-specific overrides of checked-in presets.

**Comment by vadi2** [comment]
I think `git clean` will still wipe them, and that's something I use on occasion to ensure a clean state of the repo.

**Comment by Zvicii** [comment]
run into the same issue, hope for a quick fix

**Comment by jastich** [comment]
Yes, I would like a new setting to override the `buildPresets.jobs` setting in `CMakePresets.json` to respect the `"cmake.parallelJobs"` setting in the extension.

**Comment by i-C-o-d-e-r** [comment]
I just ran into this same issue and I would like to mention that the documentation is also lacking the information that `cmake.parallelJobs` is ignored, when `CMakePresets.json` is used. I guess this is true for some other options as well.
